compiler/vm.nim

```diff
@@ -1119,13 +1119,26 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
       let newLen = regs[rb].intVal.int
       if regs[ra].node.isNil: stackTrace(c, tos, pc, errNilAccess)
       else:
+        # FIXME: this doesn't attempt to solve incomplete
+        # support of tyPtr, tyRef in VM.
+        let typ = regs[ra].node.typ.skipTypes(abstractInst+{tyRange}-{tyTypeDesc})
```

Comment:
I'm not sure if you mean extract just the
```skipTypes(abstractInst+{tyRange}-{tyTypeDesc})``` logic or the whole block into a helper proc.